plugins/CTBTriggerCandidateMaker.cpp

```diff
@@ -0,0 +1,204 @@
+/**
+ * @file CTBTriggerCandidateMaker.cpp
+ *
+ * This is part of the DUNE DAQ Application Framework, copyright 2020.
+ * Licensing/copyright details are in the COPYING file that you should have
+ * received with this code.
+ */
+
+#include "CTBTriggerCandidateMaker.hpp"
+#include "trigger/Logging.hpp"
+
+#include "appfwk/DAQModuleHelper.hpp"
+#include "iomanager/IOManager.hpp"
+#include "rcif/cmd/Nljs.hpp"
+#include "trgdataformats/Types.hpp"
+#include "trigger/TriggerCandidate_serialization.hpp"
+
+#include <bitset>
+#include <regex>
+#include <string>
+
+using dunedaq::trigger::logging::TLVL_DEBUG_ALL;
+using dunedaq::trigger::logging::TLVL_DEBUG_HIGH;
+using dunedaq::trigger::logging::TLVL_DEBUG_MEDIUM;
+using dunedaq::trigger::logging::TLVL_GENERAL;
+using dunedaq::trigger::logging::TLVL_VERY_IMPORTANT;
+
+namespace dunedaq {
+namespace trigger {
+
+CTBTriggerCandidateMaker::CTBTriggerCandidateMaker(const std::string& name)
+  : DAQModule(name)
+  , m_output_queue(nullptr)
+  , m_queue_timeout(100)
+{
+
+  register_command("conf", &CTBTriggerCandidateMaker::do_conf);
+  register_command("start", &CTBTriggerCandidateMaker::do_start);
+  register_command("stop", &CTBTriggerCandidateMaker::do_stop);
+  register_command("scrap", &CTBTriggerCandidateMaker::do_scrap);
+}
+
+std::vector<triggeralgs::TriggerCandidate>
+CTBTriggerCandidateMaker::HSIEventToTriggerCandidate(const dfmessages::HSIEvent& data)
+{
+  TLOG_DEBUG(TLVL_DEBUG_MEDIUM) << "[CTB] Converting HSI event, signal: " << data.signal_map;
+
+  std::vector<triggeralgs::TriggerCandidate> candidates;
+  std::bitset<32> bits(data.signal_map);
+  TLOG_DEBUG(TLVL_DEBUG_HIGH) << "[CTB] BITS: " << bits;
+
+  for (size_t i = 0; i < bits.size(); ++i) {
+    if (bits.test(i)) {
+
+      TLOG_DEBUG(TLVL_DEBUG_ALL) << "[CTB] this bit: " << i;
+
+      if (m_HLT_TC_map.count(i)) {
+        TLOG_DEBUG(TLVL_DEBUG_ALL) << "[CTB] TC type: " << static_cast<int>(m_HLT_TC_map[i]);
+
+        triggeralgs::TriggerCandidate candidate;
+        candidate.time_candidate = data.timestamp;
+        candidate.time_start = data.timestamp - m_time_before;
+        candidate.time_end = data.timestamp + m_time_after;
+        // candidate.detid = 1;
+        candidate.detid = data.header;
```

Comment:
I think this is probably the best way of dealing with "detid" here... `HSIEvent::header` tells us which device the event came from